CodeiumVS/Utilities/Extensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using System.Collections.Generic;
+
+namespace CodeiumVS.Utilities;
+
+internal abstract class TextViewExtension<T> where T : class
```

Comment:
This class will be the base class for anything that is associated with a `ITextView`, it provides a `Dictionary<ITextView, T> Instances` and `T? GetInstance(ITextView hostView)` method which will get an instance of the derived class that is attached to the view.